src/process.rs

```diff
@@ -94,15 +105,15 @@ pub trait AbstractProcess {
 }
 
 /// Defines a handler for a message of type `M`.
-pub trait MessageHandler<M, S = Bincode>: AbstractProcess
+pub trait MessageHandler<M, S = Bincode>: AbstractProcess<S>
```

Comment:
We can be more flexible here and don't need to match the same serializer `S` that `AbstractProcess` is using. We can have a per message/request serializer.

Reply:
I think that makes sense, though regarding this implementation:
https://github.com/lunatic-solutions/lunatic-rs/blob/b1d102cc72e0ee6882a01e8c50e195ba427505c5/src/process.rs#L554-L557
I'm not sure what to put for the `AS` in `T: MessageHandler<M, S, AS>,`, since `AS` wouldn't be tied to `Message` nor `ProcessRef`.

Reply:
I think we need it still on `ProcessRef` :(.

I imagine something like this:

```rust
impl<M, S, AS, T> Message<M, S> for ProcessRef<T, AS> 
 where 
     T: AbstractProcess<AS>, 
     T: MessageHandler<M, S>, 
```

Reply:
Though I guess you initially meant add a 3rd generic for the `AbstractProcess` serializer?
```diff
- pub trait MessageHandler<M, S = Bincode>: AbstractProcess<S>
+ pub trait MessageHandler<M, S = Bincode, AS = Bincode>: AbstractProcess<AS>
```

Reply:
I've tried to add this generic in this commit: https://github.com/lunatic-solutions/lunatic-rs/pull/64/commits/b6f68420d150a9b336d22fdd0038ded6998d5894

I'm quite amazed at how complicated the types have become at this point :disappointed: 
Especially this part:
https://github.com/lunatic-solutions/lunatic-rs/blob/b6f68420d150a9b336d22fdd0038ded6998d5894/src/process.rs#L759-L777

I don't know why it's become this complicated